### ISSUE TITLE
Potential exception when using rxjs 7 in Angular 13

### ISSUE THREAD

**Issue by cmxl** [assuncaocharles/ngx-indexed-db#295]
Environment:
- angular: 13.1.0
- rxjs: 7.4.0

Stacktrace:
```
TypeError: Cannot read properties of undefined (reading 'isStopped')
    at error (Subscriber.js:35:18)
    at IDBTransaction.error [as __zone_symbol__ON_PROPERTYerror] (ngx-indexed-db.js:106:13)
    at IDBTransaction.B (zone.js:803:43)
    at ir.invokeTask (zone.js:434:35)
    at Object.onInvokeTask (core.mjs:25437:33)
    at ir.invokeTask (zone.js:433:64)
    at ir.runTask (zone.js:205:51)
    at ir.invokeTask [as invoke] (zone.js:516:38)
    at pr (zone.js:1656:18)
    at IDBTransaction.Er (zone.js:1682:21)
```

This happens when calling `getAllKeysByIndex` and it fails.
When creating the transaction `obs.error` is directly passed on and I guess in rxjs7 the callback uses a `this` which "gets lost" due to the closure there.

See https://github.com/assuncaocharles/ngx-indexed-db/blob/c020aed2321078f8ca41d1690f7d55a16098431e/projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts#L536

I have also seens this in other places in code.
Everything here is just a guess, but maybe someone else could clarify.

I have a rather large app where I can reproduce the issue, but could not yet create a minimal repro for it.

**Comment by cmxl** [comment]
![image](https://user-images.githubusercontent.com/10906748/149129492-dac371fd-caf1-4d7a-8734-5c8eca0be013.png)

**Comment by cmxl** [comment]
directly related to #275

**Comment by ZachHaber** [comment]
I've also run into that same typeError with angular 12 and rxjs 6

**Comment by jonlighthill** [comment]
I have multiple users experiencing this as well and they are all using mobile safari 15.2 or 15.3.

I run a source map resolve and get the following, not really helpful.

`source-map resolve .\main.2918766561d1650f.js.map 1 748275
Maps to webpack:///node_modules/rxjs/_esm2015/internal/Subscriber.js:53:18 (isStopped)

        if (!this.isStopped) {`

`source-map resolve .\main.2918766561d1650f.js.map 1 1124267
Maps to webpack:///node_modules/ngx-indexed-db/fesm2015/ngx-indexed-db.mjs:106:19 (e)

            reject(e);`

`source-map resolve .\main.2918766561d1650f.js.map 1 938585
Maps to webpack:///node_modules/@angular/core/fesm2015/core.mjs:25415:43 (target)

                return delegate.invokeTask(target, task, applyThis, applyArgs);`

**Comment by ZachHaber** [comment]
I will say, that I ended up just using https://www.npmjs.com/package/idb and just using indexeddb directly instead of through Angular, and it's working pretty well for me. Sometimes framework-agnostic solutions are all you need.

**Comment by noobed7** [comment]
I have a similar issue.
When one of the storeSchema items has `unique: true` and inserting the same value throws and exception which is thrown to the console and I can't catch it inside the observable path.
For example `.pipe(catchError(x=> ...))` 
Tried to locate the reason, fix and run it locally but with no success. I'd suppose I don't know how to overcome the closures correctly 
![image](https://user-images.githubusercontent.com/35348755/159742414-15160c94-d776-413b-b9fb-5402f85b5e60.png)
I was able to handle the error in my observable, but the error persisted `TypeError: can't access property "isStopped", this is undefined` :( 
any suggestions?

**Comment by HFTSOL** [comment]
Has there been any movement on this issue?  I am seeing this issue in versions 9.3 - 10 and I'm using Angular 12 and RxJs 6.6.  I upgraded from 9.1.2 because of the "objectStore does not exists" bug and hit this one.  I need to know if it's going to be fixed in the next week or I have to remove the lib and replace it with something else.    Thanks.

**Comment by aparzi** [comment]
Hi @HFTSOL,
I hope I can take a look at the problem these days. I can't promise you anything because this is a new issue for me

**Comment by anymos** [comment]
Same here, with everything at minimum ... this lib is not production ready obviously ....

**Comment by anymos** [comment]
> I will say, that I ended up just using https://www.npmjs.com/package/idb and just using indexeddb directly instead of through Angular, and it's working pretty well for me. Sometimes framework-agnostic solutions are all you need.

Yeah I will do the same sounds a more solid option to me also

**Comment by anymos** [comment]
funny, ticket closed without any explanation,  achieved to convinced me that I take the right path. 

Nothing wrong to have a pet project like your that is not production ready, but at least mention it in the description, and as such it will help the dev regarding the javascript fatigue of having to find the right lib for the right job ...

**Comment by assuncaocharles** [comment]
@anymos closed by mistake, but please keep in mind this is a open source project maintained by people who has life, jobs and other things to do and get nothing by writing code here. It's open source, you see a hug or a reason to consider it not production ready your time will be better spent opening a PR than making sure to screaming your insatisfaction here while not paying anything to someone who has freely put some code to anyone who wants to use.

**Comment by assuncaocharles** [comment]
Were any of you able to create a minimal reproduction for the bug?

**Comment by noobed7** [comment]
Hi, 
as I wasn't sure how to give a demo, I forked your repo and pushed to a new branch

https://github.com/noobed7/ngx-indexed-db/tree/break-simulation - here
I tried to email you earlier this week, but it might have gone to spam. I'd like to propose a fix for this issue since it's a trivial js closure problem.

For example `/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts` in this file, 
this line 
`const transaction = createTransaction(db, optionsGenerator(DBMode.readwrite, storeName, obs.error));`
could be fixed as follows:
 `const transaction = createTransaction(db, optionsGenerator(DBMode.readwrite, storeName, (e) => obs.error(e)));` 

we are capturing the error callback argument with the help of a closure arrow function. This way the proper hanler is invoked and the console is clear from unhandled exceptions
